File: RedSheet.DbEntities/Models/LanguageContent.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;
using Rx.Core.Data.Attributes;
using RedSheet.DbEntities.Enums;

namespace RedSheet.DbEntities.Models
{
	[Table("LanguageContents", Schema = "dbo")]
	[Serializable]
	public partial class LanguageContent  
	{

	public string Amharic { get; set; } 

	public string Bashkir { get; set; } 
	///<summary>
    ///This is Only Used for Server Message Or Server side keys operations.
    ///</summary> 
	[MaxLength(50)]
	public string ContentType { get; set; } 

	public string English { get; set; } 
 
	[DatabaseGenerated(DatabaseGeneratedOption.Identity)] 
	[Key]
	public int LanguageContentId { get; set; } 
 
	[Required] 
	[MaxLength(50)]
	public string LanguageContentName { get; set; } 

	public string Zulu { get; set; } 
	[InverseProperty("LanguageContent")]
	public ICollection<EmailTemplateDetail> EmailTemplateDetails { get; set; } 

	[InverseProperty("LanguageContent")]
	public ICollection<ModuleContent> ModuleContents { get; set; } 


	public LanguageContent()
	{
	this.EmailTemplateDetails = new HashSet<EmailTemplateDetail>();

	this.ModuleContents = new HashSet<ModuleContent>();

    }

    }
}

File: redsheet-portal/src/app/database-models/language-content.ts
import { required, maxLength, range, nested } from '@rx/annotations';
import { EmailTemplateDetail, ModuleContent,  } from './'
export class LanguageContent {
    constructor(languageContent?: LanguageContent )  {
        let properties = [ "amharic", "bashkir", "contentType", "english", "languageContentId", "languageContentName", "zulu", "emailTemplateDetails", "moduleContents",];
        for (let property of properties)
            if (languageContent && languageContent[property])
                this[property] = languageContent[property];
    }
 
	amharic : string =   undefined;
 
	bashkir : string =   undefined;
 
    @maxLength(50)
	contentType : string =   undefined;
 
	english : string =   undefined;
 
	languageContentId : number =   0 ;
 
    @required()
    @maxLength(50)
	languageContentName : string =   undefined;
 
	zulu : string =   undefined;
	@nested(EmailTemplateDetail)
	emailTemplateDetails: EmailTemplateDetail[];

	@nested(ModuleContent)
	moduleContents: ModuleContent[];



}
